### PR TITLE
Default search context

### PR DESCRIPTION
**Pull request by Tthomas63**
Moves the default search results to the index page's context
- Add static method to `SearchView` class that returns default results
- Clean-up

### CODE REVIEW

File: src/settings/base.py [hunk 1]
```diff
@@ -207,10 +207,10 @@
 REST_FRAMEWORK = {
     'DEFAULT_AUTHENTICATION_CLASSES': (
         'rest_framework.authentication.SessionAuthentication',
-        'oauth2_provider.contrib.rest_framework.OAuth2Authentication',
+        # 'oauth2_provider.contrib.rest_framework.OAuth2Authentication',
```
Comment (ckcollab) [inline src/settings/base.py]: pull

File: src/static/riot/results.tag [hunk 2]
```diff
@@ -380,9 +383,16 @@
 
             CHAHUB.api.search(filters)
                 .done(function (data) {
+                    var tmp_results = []
+                    if (data.showing_default_results === true){
+                        tmp_results = DEFAULT_SEARCH_RESULTS
+                    }
```
Comment (ckcollab) [inline src/static/riot/results.tag]: same line as else

File: src/templates/index.html [hunk 3]
```diff
@@ -60,6 +60,16 @@
         API: "/api/v1/"
     }
     PRODUCERS = JSON.parse("{{ producer_data|escapejs }}")
+    DEFAULT_SEARCH_RESULTS = JSON.parse("{{ default_search_results|escapejs }}")
+
+    LOGIN_URL = "{% url "login" %}"
+    console.log(LOGIN_URL)
```
Comment (ckcollab) [inline src/templates/index.html]: pull

File: src/templates/registration/logged_out.html [hunk 4]
```diff
@@ -12,7 +12,7 @@ <h1>Accelerating reproducible computational research.</h1>
                 <div class="ui error message">
                     You have been logged out.
                 </div>
-                <a href="{% url "pages:home" %}" class="ui fluid blue submit labeled icon button">
+                <a href="/" class="ui fluid blue submit labeled icon button">
```
Comment (ckcollab) [inline src/templates/registration/logged_out.html]: fix (should be {% url %})

File: src/urls.py [hunk 5]
```diff
@@ -50,13 +53,15 @@ def get_context_data(self, **kwargs):
                     'name': producer.name,
                     'url': producer.url
                 })
-
         context['producer_data'] = json.dumps(context['producer_data'])
+        # Call our default sort helper function
+        context['default_search_results'] = json.dumps(SearchView._default_search())
```
Comment (ckcollab) [inline src/urls.py]: one thing to check here, what if something contains a script? when this is used on the frontend is it possible to XSS?

File: src/templates/registration/logged_out.html [hunk 6]
```diff
@@ -12,7 +12,7 @@ <h1>Accelerating reproducible computational research.</h1>
                 <div class="ui error message">
                     You have been logged out.
                 </div>
-                <a href="/" class="ui fluid blue submit labeled icon button">
+                <a href="{% url %}" class="ui fluid blue submit labeled icon button">
```
Comment (ckcollab) [inline src/templates/registration/logged_out.html]: should be like `{% url "home" %}`

**Review by ckcollab — CHANGES_REQUESTED** [review]
remove cruft?

File: src/urls.py (hunk 5 above)
Comment (Tthomas63) [reply src/urls.py]: Are you saying like in the competition HTML? I do not think that is possible... Would be interesting to look into though

**Pull request closed without merging**